new_course/1_programming_fundementals/methods.md

```diff
@@ -1,62 +1,268 @@
 
-This is what methods are for. They allow you to wrap code in a name which you
-can then use where you need that code to be run in your programs.
+This concept allows for what programmers refer to as the *D.R.Y.* approach to programming *-Don't Repeat Yourself-*. Recycling parts of code that would otherwise need to be repeated, often numerous times in several places, *can* make your program more readable and more manageable.
 
 In this lesson we are going to deconstruct what methods are, their behaviour, and how they are used.
 
 ## Learning Outcomes
 *Look through these now and then use them to test yourself after doing the assignment*
 
 * You understand how to use Ruby's built in methods on objects
-* You know how to create your own custom methods
-* You know how to *call* your own methods
-* You understand the difference between a explicit and implicit return in your methods
-* You understand the difference between `puts` and `return`
+* How would you create your own custom method?
+* How would you *call* your own method?
+* How do you pass variables to your method?
+* How do you define default parameters for your method?
+* What is the difference between an *explicit* and *implicit* return in your methods?
+* What is the difference between `puts` and `return`?
+* How do you chain multiple methods together?
+* Give an example of a valid method name and an invalid method name.
+* What does "snake case" naming mean?
+* What are some of Ruby's reserved words?
+* What do you call a method that returns `true` or `false`? How are they named?
+* What do bang methods do? How are they conventionally named?
 
 ## Ruby's Built in Methods
-* explain what they are, you've already been using them.
+
+One of Ruby's great advantages for new programmers is the large number of built-in methods it includes. You've been using many of them already, probably without even realizing it. Over the course of your learning so far, you have modified strings and other objects in various ways. Each of these operations you have performed has been a method. These are most often applied by adding `.method_name` after the object you are modifying.
+
+```ruby
+"anything".reverse
+```
+
+In this case, `reverse` is a method Ruby has built in for strings (and some others data types).
+
+`print` and `puts` are also methods. They are just called in a different way, by typing their name and passing them arguments.
+
+```ruby
+puts "anything" #=> anything
+```
+
+It's worth noting that in most languages, arguments are passed to methods by wrapping them in parentheses `()`. In Ruby, the parentheses are optional. So the above code is the same as `puts("anything")`, which may look more familiar if you are used to another programming language, such as JavaScript.
```

Comment:
I'm not sure if it's worth mentioning at this stage but something I've seen trip learners up (myself included) is that parens are required in some instances when using puts because of the precedence of method calls. I need to find a good example though